getting-started/installation/general-installation/installation-community-containerized.markdown

```diff
@@ -0,0 +1,249 @@
+```
+
+## Preparing CFEngine host in container
+The procedure is similar to the cfengine-hub deployment, with changes to the name of the container for identification.
+
+```command
+$ docker run --privileged -dit --name=cfengine-host registry.access.redhat.com/ubi9-init /usr/sbin/init
+```
+
+Rest of the installation steps remain the same as for **cfengine-hub**.
+
+### Bootstrap cfengine-host to the policy server container.
+Find ip of **cfengine-hub**:
+
+```command
+$ CFENGINE_HUB_IP=$(docker exec cfengine-hub bash -c "ip -4 -o addr show eth0 | awk '{print \$4}' | cut -d'/' -f1")
+```
+
+Bootstrap cfengine-host to cfengine-hub:
+
+```command
+$ docker exec cfengine-host bash -c "/usr/local/sbin/cf-agent --bootstrap ${CFENGINE_HUB_IP}"
+```
+
+## Using docker compose
+### Preparing container image for CFEngine
+Create a `Dockerfile` with following contents:
+
+```cfengine3
+FROM registry.access.redhat.com/ubi9-init:latest
+LABEL container ubi9-init_cfengine-community-3.24.0.1
+
+RUN dnf -y update \
+&& dnf -y install bind-utils iproute procps-ng \
+&& dnf -y install https://cfengine-package-repos.s3.amazonaws.com/community_binaries/Community-3.24.0/agent_rhel9_x86_64/cfengine-community-3.24.0-1.el9.x86_64.rpm
+
+HEALTHCHECK --interval=5s --timeout=15s --retries=3 \
+    CMD /usr/local/sbin/cf-agent --self-diagnostics || exit 1
+
+ENTRYPOINT ["/usr/sbin/init"]
+```
+
+Build the docker image based on above Dockerfile:
+
+```command
+$ docker build -t cfengine:3.24.0-1 -f Dockerfile .
+```
+
+Verify created image:
+
+```command
+$ docker image ls cfengine
+```
+```output
+REPOSITORY   TAG        IMAGE ID       CREATED             SIZE
+cfengine     3.24.0-1   <IMAGE_ID>     About an hour ago   302MB
+```
+
+**Note**: One can optionally skip step 2 and 3 above, if planning to use `docker compose`.
+
+### Using docker compose service
+Create a `compose.yml` file with following contents:
+
+```cfengine3
+name: cfengine-demo
+
+services:
+    cfengine-hub:
+        container_name: cfengine-hub
+        image: cfengine:3.24.0-1
+        build:
+        context: .
+        dockerfile: Dockerfile
+        privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+            /usr/local/sbin/cf-agent --bootstrap \$(ip -4 -o addr show eth0 | awk '{print \$4}' | cut -d'/' -f1)
+        networks:
+        - control-plane
+
+    cfengine-host:
+        image: cfengine:3.24.0-1
+        build:
+        context: .
+        dockerfile: Dockerfile
+        privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+            /usr/local/sbin/cf-agent --bootstrap \$(dig +short cfengine-hub|tr -d [:space:])
+        networks:
+        - control-plane
+        depends_on:
+        cfengine-hub:
+            condition: service_healthy
+            required: true
+
+networks:
+    control-plane:
+```
+
+Start service cfengine-demo
+
+```command
+$ docker compose -f compose.yaml --verbose up -d
```

Comment:
```suggestion
docker compose -f compose.yaml --verbose up -d
```

Reply:
I get an error running this.

```
validating /tmp/test/compose.yaml: services.cfengine-host.depends_on must be a list
```

Reply:
Yes, there was an indentation issue in the `config.yaml`. I've fixed it.